File: typescript/darts/darts.ts
export const score = (x: number, y: number): number => {
  if (outOfDartboard(x, y)) return 0
  if (inInnerCircle(x, y)) return 10
  if (inMiddleCircle(x, y)) return 5
  if (inOuterCircle(x, y)) return 1
  return 0 
}

const outOfDartboard = (x: number, y: number): boolean => distance(x, y) > 10
const inInnerCircle = (x: number, y: number): boolean => distance(x, y) <= 1
const inMiddleCircle = (x: number, y: number): boolean => distance(x, y) <= 5
const inOuterCircle = (x: number, y: number): boolean => distance(x, y) <= 10
const distance = (x: number, y: number): number => Math.sqrt(x * x + y * y)

File: haskell/darts/src/Darts.hs
module Darts (score) where

score :: Float -> Float -> Int
score x y
  | r <= 1 = 10
  | r <= 5 = 5
  | r <= 10 = 1
  | otherwise = 0
  where
    r = sqrt (x ^ 2 + y ^ 2)
